src/order/category/ωCPO.lean

```diff
@@ -0,0 +1,269 @@
+/-
+Copyright (c) 2020 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+-/
+
+import order.omega_complete_partial_order
+import order.category.Preorder
+
+/-! # Category of types with a omega complete partial order
+
+In this file, we bundle the class `omega_complete_partial_order` into a
+concrete category and prove that continuous functions also form
+a `omega_complete_partial_order`.
+
+## Main definitions
+
+ * `continuous_hom` (with notation →𝒄)
+   * an instance of `omega_complete_partial_order (α →𝒄 β)`
+ * `continuous_hom.of_fun`
+ * `continuous_hom.of_mono`
+ * continuous functions:
+   * `id`
+   * `ite`
+   * `const`
+   * `roption.bind`
+   * `roption.map`
+   * `roption.seq`
+ * `ωCPO`
+   * an instance of `category` and `concrete_category`
+
+ -/
+
+open category_theory
+
+universes u v
+
+namespace omega_complete_partial_order
+
+variables (α : Type*) (β : Type*) {γ : Type*} {φ : Type*}
+variables [omega_complete_partial_order α]
+variables [omega_complete_partial_order β]
+variables [omega_complete_partial_order γ]
+variables [omega_complete_partial_order φ]
+
+section old_struct
+
+set_option old_structure_cmd true
+
+/-- A monotone function is continuous if it preserves the supremum of chains -/
```

Comment:
The relationship between this predicate and `continuous f` from topology should be clarified. Is it the same thing? Is there an implication? Are they incompatible?

Reply:
(Just to be clear, I don't want to to respond to me saying how they are the same/different, I want you to write that response in the doc.)

Reply:
I would say that they are incompatible but I haven't studied the matter enough to assert it in the documentation. I could say that they are distinct notions.

Reply:
Update: I just read about Scott Topology: https://ncatlab.org/nlab/show/Scott+topology

It is defined in relation with `DCPOs` but I think the same can be built with `ωCPOs`

Reply:
Here is a longer answer to this question: https://github.com/leanprover-community/mathlib/tree/unbounded-recursion-scott-topology. I don't know what it's useful for but there is a connection between topology and CPOs